src/types/UserChangemakerPermission.ts

```diff
@@ -10,6 +10,7 @@ interface UserChangemakerPermission {
 	readonly changemakerId: number;
 	readonly createdBy: KeycloakUserId;
 	readonly createdAt: string;
+	readonly notAfter?: string;
```

Comment:
We want `notAfter` to always be present, don't we? Just sometimes unset? I know you went to a lot of trouble wrangling types for `Writable`!

Or, maybe we don't need to read `notAfter` outside of the database at all yet, and we can just omit it from the type definition and the SQL query.

Reply:
Side question: TS guidelines say "use undefined, not null" -- https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

What do you think about all that?  This wouldn't change your request of having it be required (just possibly undefined) e.g. `notAfter: string | undefined`

as for the other points I'll get back to ya, I did circle around some of those and all three were implemented at one point but that doesn't mean I disagree with them.

Reply:
I mean, the top of that page is pretty explicit about its target audience! :stuck_out_tongue: 

I think the semantics of `null` more closely align with what we're intending to convey, and is analogous to the SQL `NULL` that's backing it: it's not that we have this variable that hasn't been initialized yet; we know the value of `notAfter`, and it is explicitly missing.

Reply:
Oh right, so it is!  I have to re-remember that every year or two